tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/GenerateRepositoryFacadeMojo.java

```diff
@@ -579,6 +605,31 @@ private void addP2Stats(File p2repository) throws FileNotFoundException, IOExcep
 		contentStream.close();
 		outContentStream.closeEntry();
 		outContentStream.close();
+
+		// JBDS-3929 overwrite the artifacts.xml.xz file too
+		// see also https://bugs.eclipse.org/bugs/show_bug.cgi?id=464614
+		//getLog().debug("delete old artifacts.xml.xz");
+		File artifactsXmlXz = new File(p2repository,"artifacts.xml.xz");
+		FileUtils.forceDelete(artifactsXmlXz);
```

Comment:
Should this code for .xz'en the file be put in a separate method instead of duplicating that many lines ?